### ISSUE TITLE
Controller never succeeds, and turns around at the end of the path

### ISSUE THREAD

**Issue by rokusottervanger** [nobleo/path_tracking_pid#10]
I'm sending a plan produced by the generic `global_planner/GlobalPlanner`, and this happens:
![never_succeeds](https://user-images.githubusercontent.com/8997639/150363119-3a48fa4f-9105-4dbe-bb28-641882204715.gif)
After this, it slowly follows the path back to the start.

**Comment by Timple** [comment]
Hmm. I'm quite sure a straight line is part of the testcases.

Perhaps we stop the test when reaching the end, not evaluating anything after that?

**Comment by rokusottervanger** [comment]
Could this be because the `min_x_vel` is nonzero? But still, why would it turn around and trace the plan back to the start, start over and do it again?

**Comment by eborghi10** [comment]
This seems to happen because in the local planner there's a method call that is always return false:

```C++
bool TrackingPidLocalPlanner::isGoalReached() const
{
  // Return reached boolean, but never succeed when we're preempting
  return pid_controller_.isEndReached() && !cancel_in_progress_;
}
```

If we inspect the call to `isEndReached` in the `path_tracking_pid::Controller` class:

```C++
bool isEndReached() const { return controller_state_.end_reached; }
```

If you check the controller.cpp file, the variable `end_reached` is never set to true.

**Comment by eborghi10** [comment]
I experienced the same issue with my robot. The logic that decides whether the goal is reached considers the boolean variable `end_reached` from src/controller.cpp.

```C++
if (
    (distance_to_goal == 0.0 && target_end_x_vel >= VELOCITY_EPS) ||
    (controller_state_.end_phase_enabled && new_x_vel >= target_end_x_vel - VELOCITY_EPS &&
     new_x_vel <= target_end_x_vel + VELOCITY_EPS)) {
    controller_state_.end_reached = true;
```

As we can see, there are two different conditions where this is possible.

If we analyze the first one, my setup wasn't stopping because the target end forward velocity was too small (and less than `VELOCITY_EPS`):

```
[DEBUG] [1696974553.837662962, 19.042000000]: distance_to_goal: 0.000000
[DEBUG] [1696974553.837675145, 19.042000000]: target_end_x_vel: 0.000000
```

This situation is mentioned in the code:

> // Warning! If target_end_x_vel == 0 and min_vel = 0 then the robot might not reach end pose

On the other hand, the second condition checks that the final velocity is reached. In this case, if the parameters `abs_minimum_x_vel` and `target_end_x_vel` are larger than `VELOCITY_EPS`, the robot won't ever meet the condition.

In summary, the solution that I found was to set the parameters like this:

```YAML
target_end_x_vel: 0.0
abs_minimum_x_vel: 0.0
```
